// next@14.2.28/dist/shared/lib/dynamic/index.js
import { __module as a } from "../../../../../../../../../_virtual/dynamic/index.js";
import "../../../../../../../@swc_helpers@0.5.5/node_modules/@swc/helpers/cjs/_interop_require_default/index.js";
import S from "react/jsx-runtime";
import "react";
import { __require as q } from "../loadable.shared-runtime/index.js";
import { __exports as P } from "../../../../../../../../../_virtual/_interop_require_default/index.js";
(function(s, t) {
  Object.defineProperty(t, "__esModule", {
    value: !0
  });
  function c(r, n) {
    for (var l in n) Object.defineProperty(r, l, {
      enumerable: !0,
      get: n[l]
    });
  }
  c(t, {
    /**
    * This function lets you dynamically import a component.
    * It uses [React.lazy()](https://react.dev/reference/react/lazy) with [Suspense](https://react.dev/reference/react/Suspense) under the hood.
    *
    * Read more: [Next.js Docs: `next/dynamic`](https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading#nextdynamic)
    */
    default: function() {
      return b;
    },
    noSSR: function() {
      return f;
    }
  });
  const _ = P, o = S, m = /* @__PURE__ */ _._(q()), p = typeof window > "u";
  function d(r) {
    return {
      default: (r == null ? void 0 : r.default) || r
    };
  }
  function f(r, n) {
    if (delete n.webpack, delete n.modules, !p)
      return r(n);
    const l = n.loading;
    return () => /* @__PURE__ */ (0, o.jsx)(l, {
      error: null,
      isLoading: !0,
      pastDelay: !1,
      timedOut: !1
    });
  }
  function b(r, n) {
    let l = m.default, e = {
      // A loading component is not required, so we default it
      loading: (y) => {
        let { error: u, isLoading: v, pastDelay: g } = y;
        if (!g) return null;
        if (process.env.NODE_ENV !== "production") {
          if (v)
            return null;
          if (u)
            return /* @__PURE__ */ (0, o.jsxs)("p", {
              children: [
                u.message,
                /* @__PURE__ */ (0, o.jsx)("br", {}),
                u.stack
              ]
            });
        }
        return null;
      }
    };
    r instanceof Promise ? e.loader = () => r : typeof r == "function" ? e.loader = r : typeof r == "object" && (e = {
      ...e,
      ...r
    }), e = {
      ...e,
      ...n
    };
    const i = e.loader, j = () => i != null ? i().then(d) : Promise.resolve(d(() => null));
    return e.loadableGenerated && (e = {
      ...e,
      ...e.loadableGenerated
    }, delete e.loadableGenerated), typeof e.ssr == "boolean" && !e.ssr ? (delete e.webpack, delete e.modules, f(l, e)) : l({
      ...e,
      loader: j
    });
  }
  (typeof t.default == "function" || typeof t.default == "object" && t.default !== null) && typeof t.default.__esModule > "u" && (Object.defineProperty(t.default, "__esModule", { value: !0 }), Object.assign(t.default, t), s.exports = t.default);
})(a, a.exports);
var E = a.exports;
export {
  E as d
};
