// axios@1.10.0/lib/utils/index.js
import D from "../helpers/bind/index.js";
const { toString: R } = Object.prototype, { getPrototypeOf: h } = Object, { iterator: y, toStringTag: x } = Symbol, w = /* @__PURE__ */ ((e) => (t) => {
  const r = R.call(t);
  return e[r] || (e[r] = r.slice(8, -1).toLowerCase());
})(/* @__PURE__ */ Object.create(null)), l = (e) => (e = e.toLowerCase(), (t) => w(t) === e), F = (e) => (t) => typeof t === e, { isArray: a } = Array, p = F("undefined");
function T(e) {
  return e !== null && !p(e) && e.constructor !== null && !p(e.constructor) && i(e.constructor.isBuffer) && e.constructor.isBuffer(e);
}
const S = l("ArrayBuffer");
function M(e) {
  let t;
  return typeof ArrayBuffer < "u" && ArrayBuffer.isView ? t = ArrayBuffer.isView(e) : t = e && e.buffer && S(e.buffer), t;
}
const N = F("string"), i = F("function"), B = F("number"), g = (e) => e !== null && typeof e == "object", k = (e) => e === !0 || e === !1, m = (e) => {
  if (w(e) !== "object")
    return !1;
  const t = h(e);
  return (t === null || t === Object.prototype || Object.getPrototypeOf(t) === null) && !(x in e) && !(y in e);
}, I = l("Date"), U = l("File"), V = l("Blob"), _ = l("FileList"), q = (e) => g(e) && i(e.pipe), H = (e) => {
  let t;
  return e && (typeof FormData == "function" && e instanceof FormData || i(e.append) && ((t = w(e)) === "formdata" || // detect form-data instance
  t === "object" && i(e.toString) && e.toString() === "[object FormData]"));
}, z = l("URLSearchParams"), [J, K, $, W] = ["ReadableStream", "Request", "Response", "Headers"].map(l), G = (e) => e.trim ? e.trim() : e.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, "");
function d(e, t, { allOwnKeys: r = !1 } = {}) {
  if (e === null || typeof e > "u")
    return;
  let n, s;
  if (typeof e != "object" && (e = [e]), a(e))
    for (n = 0, s = e.length; n < s; n++)
      t.call(null, e[n], n, e);
  else {
    const o = r ? Object.getOwnPropertyNames(e) : Object.keys(e), c = o.length;
    let f;
    for (n = 0; n < c; n++)
      f = o[n], t.call(null, e[f], f, e);
  }
}
function E(e, t) {
  t = t.toLowerCase();
  const r = Object.keys(e);
  let n = r.length, s;
  for (; n-- > 0; )
    if (s = r[n], t === s.toLowerCase())
      return s;
  return null;
}
const u = typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : global, L = (e) => !p(e) && e !== u;
function O() {
  const { caseless: e } = L(this) && this || {}, t = {}, r = (n, s) => {
    const o = e && E(t, s) || s;
    m(t[o]) && m(n) ? t[o] = O(t[o], n) : m(n) ? t[o] = O({}, n) : a(n) ? t[o] = n.slice() : t[o] = n;
  };
  for (let n = 0, s = arguments.length; n < s; n++)
    arguments[n] && d(arguments[n], r);
  return t;
}
const Q = (e, t, r, { allOwnKeys: n } = {}) => (d(t, (s, o) => {
  r && i(s) ? e[o] = D(s, r) : e[o] = s;
}, { allOwnKeys: n }), e), X = (e) => (e.charCodeAt(0) === 65279 && (e = e.slice(1)), e), Y = (e, t, r, n) => {
  e.prototype = Object.create(t.prototype, n), e.prototype.constructor = e, Object.defineProperty(e, "super", {
    value: t.prototype
  }), r && Object.assign(e.prototype, r);
}, Z = (e, t, r, n) => {
  let s, o, c;
  const f = {};
  if (t = t || {}, e == null) return t;
  do {
    for (s = Object.getOwnPropertyNames(e), o = s.length; o-- > 0; )
      c = s[o], (!n || n(c, e, t)) && !f[c] && (t[c] = e[c], f[c] = !0);
    e = r !== !1 && h(e);
  } while (e && (!r || r(e, t)) && e !== Object.prototype);
  return t;
}, j = (e, t, r) => {
  e = String(e), (r === void 0 || r > e.length) && (r = e.length), r -= t.length;
  const n = e.indexOf(t, r);
  return n !== -1 && n === r;
}, v = (e) => {
  if (!e) return null;
  if (a(e)) return e;
  let t = e.length;
  if (!B(t)) return null;
  const r = new Array(t);
  for (; t-- > 0; )
    r[t] = e[t];
  return r;
}, ee = /* @__PURE__ */ ((e) => (t) => e && t instanceof e)(typeof Uint8Array < "u" && h(Uint8Array)), te = (e, t) => {
  const n = (e && e[y]).call(e);
  let s;
  for (; (s = n.next()) && !s.done; ) {
    const o = s.value;
    t.call(e, o[0], o[1]);
  }
}, ne = (e, t) => {
  let r;
  const n = [];
  for (; (r = e.exec(t)) !== null; )
    n.push(r);
  return n;
}, re = l("HTMLFormElement"), se = (e) => e.toLowerCase().replace(
  /[-_\s]([a-z\d])(\w*)/g,
  function(r, n, s) {
    return n.toUpperCase() + s;
  }
), b = (({ hasOwnProperty: e }) => (t, r) => e.call(t, r))(Object.prototype), oe = l("RegExp"), P = (e, t) => {
  const r = Object.getOwnPropertyDescriptors(e), n = {};
  d(r, (s, o) => {
    let c;
    (c = t(s, o, e)) !== !1 && (n[o] = c || s);
  }), Object.defineProperties(e, n);
}, ie = (e) => {
  P(e, (t, r) => {
    if (i(e) && ["arguments", "caller", "callee"].indexOf(r) !== -1)
      return !1;
    const n = e[r];
    if (i(n)) {
      if (t.enumerable = !1, "writable" in t) {
        t.writable = !1;
        return;
      }
      t.set || (t.set = () => {
        throw Error("Can not rewrite read-only method '" + r + "'");
      });
    }
  });
}, ce = (e, t) => {
  const r = {}, n = (s) => {
    s.forEach((o) => {
      r[o] = !0;
    });
  };
  return a(e) ? n(e) : n(String(e).split(t)), r;
}, le = () => {
}, fe = (e, t) => e != null && Number.isFinite(e = +e) ? e : t;
function ue(e) {
  return !!(e && i(e.append) && e[x] === "FormData" && e[y]);
}
const ae = (e) => {
  const t = new Array(10), r = (n, s) => {
    if (g(n)) {
      if (t.indexOf(n) >= 0)
        return;
      if (!("toJSON" in n)) {
        t[s] = n;
        const o = a(n) ? [] : {};
        return d(n, (c, f) => {
          const A = r(c, s + 1);
          !p(A) && (o[f] = A);
        }), t[s] = void 0, o;
      }
    }
    return n;
  };
  return r(e, 0);
}, pe = l("AsyncFunction"), de = (e) => e && (g(e) || i(e)) && i(e.then) && i(e.catch), C = ((e, t) => e ? setImmediate : t ? ((r, n) => (u.addEventListener("message", ({ source: s, data: o }) => {
  s === u && o === r && n.length && n.shift()();
}, !1), (s) => {
  n.push(s), u.postMessage(r, "*");
}))(`axios@${Math.random()}`, []) : (r) => setTimeout(r))(
  typeof setImmediate == "function",
  i(u.postMessage)
), me = typeof queueMicrotask < "u" ? queueMicrotask.bind(u) : typeof process < "u" && process.nextTick || C, ye = (e) => e != null && i(e[y]), Fe = {
  isArray: a,
  isArrayBuffer: S,
  isBuffer: T,
  isFormData: H,
  isArrayBufferView: M,
  isString: N,
  isNumber: B,
  isBoolean: k,
  isObject: g,
  isPlainObject: m,
  isReadableStream: J,
  isRequest: K,
  isResponse: $,
  isHeaders: W,
  isUndefined: p,
  isDate: I,
  isFile: U,
  isBlob: V,
  isRegExp: oe,
  isFunction: i,
  isStream: q,
  isURLSearchParams: z,
  isTypedArray: ee,
  isFileList: _,
  forEach: d,
  merge: O,
  extend: Q,
  trim: G,
  stripBOM: X,
  inherits: Y,
  toFlatObject: Z,
  kindOf: w,
  kindOfTest: l,
  endsWith: j,
  toArray: v,
  forEachEntry: te,
  matchAll: ne,
  isHTMLForm: re,
  hasOwnProperty: b,
  hasOwnProp: b,
  // an alias to avoid ESLint no-prototype-builtins detection
  reduceDescriptors: P,
  freezeMethods: ie,
  toObjectSet: ce,
  toCamelCase: se,
  noop: le,
  toFiniteNumber: fe,
  findKey: E,
  global: u,
  isContextDefined: L,
  isSpecCompliantForm: ue,
  toJSONObject: ae,
  isAsyncFn: pe,
  isThenable: de,
  setImmediate: C,
  asap: me,
  isIterable: ye
};
export {
  Fe as default
};
